trin-execution/src/utils.rs

```diff
@@ -8,15 +20,11 @@ pub fn full_nibble_path_to_address_hash(key_path: &[u8]) -> B256 {
         );
     }
 
-    let mut raw_address_hash = vec![];
-    for i in 0..key_path.len() {
-        if i % 2 == 0 {
-            raw_address_hash.push(key_path[i] << 4);
-        } else {
-            raw_address_hash[i / 2] |= key_path[i];
-        }
-    }
-    B256::from_slice(&raw_address_hash)
+    B256::from_slice(&compress_nibbles(key_path))
+}
+
+pub fn partial_nibble_path_to_right_padded_b256(partial_nibble_path: &[u8]) -> B256 {
```

Comment:
Not a big fan of converting them to `Vec` and then to `B256`.

How about:
```rust
pub fn nibbles_to_right_padded_b256(nibbles: &[u8]) -> B256 {
    let mut result = B256::ZERO;
    for (i, nibble) in nibbles.iter().enumerate() {
        if i % 2 == 0 {
            result[i / 2] |= nibble << 4;
        } else {
            result[i / 2] |= nibble;
        };
    }
    result
}
```

We also don't need `compress_nibbles` and we can modify `full_nibble_path_to_address_hash` to:
```rust
pub fn full_nibble_path_to_address_hash(key_path: &[u8]) -> B256 {
    if key_path.len() != 64 {
        panic!(
            "Key path should always be 64 bytes long: {}",
            key_path.len()
        );
    }

    nibbles_to_right_padded_b256(key_path)
}
```